addons/mail/static/src/new/core_ui/thread.js

```diff
@@ -42,7 +42,11 @@ export class Thread extends Component {
     setup() {
         this.messaging = useMessaging();
         this.store = useStore();
-        this.state = useState({ isReplyingTo: false });
+        this.state = useState({
+            isReplyingTo: false,
+            hasLoadingFailed: false,
```

Comment:
I feel like `hasLoadingFailed` should be on thread model itself, and be handled in the various fetch/request methods directly, rather than each caller having to handle it separately.